### PR TITLE
Porting crm_claim_rma to Odoo 8

### PR DESCRIPTION
**Pull request by mistotebe**
This is a proposal for crm_claim_rma as outlined in #20, it depends on #17 and #23.

The picking creation might be slightly too naive, but it works for the stock set up I tested and I would need some help improving it if there are scenarios where the pickings fail to be created appropriately.

### CODE REVIEW

File: crm_claim_rma/__openerp__.py [hunk 1]
```diff
@@ -66,6 +66,7 @@
  * Joel Grand-Guillaume <joel.grandguillaume@camptocamp.com>
  * Guewen Baconnier <guewen.baconnier@camptocamp.com>
  * Yannick Vaucher <yannick.vaucher@camptocamp.com>
+ * Ondřej Kuzník <ondrej.kuznik@credativ.co.uk>
```
Comment (yvaucher) [inline crm_claim_rma/__openerp__.py]: Please move description in `README.rst` here is the template: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: crm_claim_rma/crm_claim_rma_view.xml [hunk 2]
```diff
@@ -217,32 +215,31 @@
...
                                 </sheet>
                             </form>
                         </field>
                     </group>
-                    <group col="4" colspan="4" attrs="{'invisible':[('state', '&lt;&gt;','open')]}">
+                    <group col="4" colspan="4">
                         <separator string="Action" colspan="4" />
                         <button name="%(action_claim_picking_in)d"
-                                string="New Products Return" states="open"
+                                string="New Products Return"
```
Comment (yvaucher) [inline crm_claim_rma/crm_claim_rma_view.xml]: Why do you change visibility of those buttons?

File: crm_claim_rma/tests/test_lp_1282584.py [hunk 3]
```diff
@@ -78,7 +73,7 @@ def test_00(self):
 
         res = self.WizardMakePicking.action_create_picking(
             cr, uid, [wizard_id], context=wiz_context)
-        self.assertEquals(res.get('res_model'), 'stock.picking.in',
+        self.assertEquals(res.get('res_model'), 'stock.picking',
                           "Wrong model defined")
```
Comment (yvaucher) [inline crm_claim_rma/tests/test_lp_1282584.py]: This check seems meaningless to me. Here we tested the wizard was opening the right view based on model.
We cannot rely on model anymore.

File: crm_claim_rma/tests/test_lp_1282584.py [hunk 4]
```diff
@@ -22,11 +22,6 @@
 
 
 class test_lp_1282584(common.TransactionCase):
-    """ Test wizard open the right type of view
-
-    The wizard can generate picking.in and picking.out
-    Let's ensure it open the right view for each picking type
-    """
```
Comment (yvaucher) [inline crm_claim_rma/tests/test_lp_1282584.py:29]: Why removing the description and not fixing the test ?

File: product_warranty/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    'data': [
+        'security/ir.model.access.csv',
+        'res_company_view.xml',
+        'product_warranty_view.xml',
+    ],
+    'demo_xml': [],
+    'test': [],
+    'installable': True,
+    'active': False,
+    'certificate': '',
```
Comment (yvaucher) [inline product_warranty/__openerp__.py]: certifitate key and active key can be removed

File: product_warranty/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Product warranty',
+    'version': '1.0',
+    'category': 'Generic Modules/Product',
+    'description': """
+Product Warranty
```
Comment (yvaucher) [inline product_warranty/__openerp__.py]: Please also move this in a `README.rst` file

File: crm_claim_rma/crm_claim_rma_view.xml (hunk 2 above)
Comment (mistotebe) [inline crm_claim_rma/crm_claim_rma_view.xml]: State field does not exist on crm.claim anymore.

Comment (yvaucher) [inline crm_claim_rma/crm_claim_rma_view.xml]: Ok thanks I missed that

**Comment by yvaucher** [comment]
Thanks it looks good to me, one last thing is that test to be modified:

You can remove test test_lp_1282584, AFAIK there is now only a single view for incoming and outgoing pickings. So no need to check which view will be openned anymore.

You might want to convert it to testing the wizard opens. So we keep the same coverage. However, reference to launchpad bug 1282584 https://bugs.launchpad.net/openerp-rma/+bug/1282584  can be removed as picking model changed in 8.0.

**Comment by mistotebe** [comment]
Unit tests are ported now and the CLA for @craiggowing should be sent over soon (we sent the signed CLAs for all credativ people over a year ago, but it must have been lost since).

**Comment by yvaucher** [comment]
Thanks for the migration of tests :+1:

**Comment by pedrobaeza** [comment]
I say the same as in the other PRs: instead of creating and managing a code for crm.claim, you should use the module _crm_claim_code_ that already handle that (https://github.com/OCA/crm/tree/8.0/crm_claim_code). Put it as dependency and remove it from your code, and we will have a lighter module.

**Comment by mistotebe** [comment]
@pedrobaeza You could have commented on #17 which this one depends on (as stated in #20) that crm_claim_code had been merged already. I will take a look at what is provided and propose any changes if necessary (I suspect the uniqueness constraint in crm_claim_code is still too fragile and incompatible with 7.0 databases for being too strict).

**Comment by aristobulo-eezee** [comment]
@pedrobaeza @gurneyalex @mistotebe I already included crm_claim_code in our PR, what I propose to you (if you agree of course) is to rebase our branch to this one and include all @mistotebe changes and ours, what do you think?

I think that it could a good starting point, since both PR are based in original crm_claim_rma features

**Comment by pedrobaeza** [comment]
Well, it seems that Vauxoo's branch contains improvements, so it's better to coordinate with them the merge.

**Comment by nhomar** [comment]
@aristobulo-eezee 

I will check your migration to validate everything is running as expected, that was one of the modules we touched less.

I will be back soon with the feedback to either join your PR to ours and make all compatible or give you specific feedback.

Regards.

**Comment by nhomar** [comment]
I will close this PR in favor of https://github.com/OCA/rma/pull/32 which is totally migrated to new Api, if somebody opose scream aloud,

**Comment by mistotebe** [comment]
@nhomar
Have you migrated the additional features as well? (Reporting fields, etc.)

**Comment by nhomar** [comment]
You can have all modules migrated here:

https://github.com/vauxoo/rma

You can follow the squashing to make the PR module per module here:

https://github.com/OCA/rma/pull/26

I think in no more than 2 weeks we will have everything merged.

Regards.

**Comment by mistotebe** [comment]
What I mean is that there are usability and reporting enhancements in this module that were not present in 7.0 version, an incomplete list:
- uom is tracked on the claim line
- when creating a claim line manually, the value is taken from cost price
- value of a line+claim is tracked

**Comment by nhomar** [comment]
2015-08-05 5:14 GMT-05:00 mistotebe notifications@github.com:

> What I mean is that there are usability and reporting enhancements in this
> module that were not present in 7.0 version, an incomplete list:
> - uom is tracked on the claim line
> - when creating a claim line manually, the value is taken from cost
>   price
> - value of a line+claim is tracked
> 
> I will give a check on this and come back to you tomorrow with the
> comments.....

## 

---

Saludos Cordiales

CEO at Vauxoo https://www.vauxoo.com Odoo's Gold Partner.

[image: --]
Nhomar Hernandez
[image: http://]about.me/nhomar
http://about.me/nhomar?promo=email_sig

**Pull request closed without merging**